tad/cli/check_state.py

```diff
@@ -0,0 +1,129 @@
+    #       How are we going to process non-assessment-like instruments?
+    if system_card.assessments:
+        return next((assessment for assessment in system_card.assessments if assessment.urn == urn), None)
+    return None
+
+
+def get_task_timestamp_from_assessment_card(task_urn: str, assessment_card: AssessmentCard) -> datetime | None:
+    for content in assessment_card.contents:
+        if content.urn == task_urn and content.timestamp:
+            return content.timestamp
+    return None
+
+
+def get_next_tasks_per_instrument(instrument: Instrument, assessment_card: AssessmentCard | None) -> dict[str, Any]:
+    return_value: dict[str, Any] = {
+        "instrument_urn": instrument.urn,
+        "last_modified_date": datetime(1970, 1, 1, tzinfo=timezone.utc),  # noqa UP017
+        "tasks_per_lifecycle": defaultdict(list[InstrumentTask]),
+    }
+
+    for instrument_task in instrument.tasks:
+        timestamp = (
+            get_task_timestamp_from_assessment_card(instrument_task.urn, assessment_card) if assessment_card else None
+        )
+
+        if timestamp and timestamp >= return_value["last_modified_date"]:
+            return_value["last_modified_date"] = timestamp
+        else:
+            lifecycle_order = get_first_lifecycle_idx(instrument_task.lifecycle)
+            return_value["tasks_per_lifecycle"][lifecycle_order].append(instrument_task)
+    return return_value
+
+
+def get_all_next_tasks(instruments: dict[str, Instrument], system_card: SystemCard) -> list[dict[str, Any]]:
+    next_tasks_per_instruments: list[dict[str, Any]] = []
+    for urn, instrument in instruments.items():
+        instrument_result_from_system_card = get_instrument_result_from_system_card(urn, system_card)
+        next_tasks_per_instruments.append(get_next_tasks_per_instrument(instrument, instrument_result_from_system_card))
+    next_tasks_per_instruments = sorted(next_tasks_per_instruments, key=lambda i: i["last_modified_date"], reverse=True)
+    return next_tasks_per_instruments
+
+
+@click.command()
+@click.argument("urns", nargs=-1)
+@click.argument("system_card_path", type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path))
+def get_tasks_by_priority(urns: list[str], system_card_path: Path) -> None:
+    try:
+        system_card = get_system_card(system_card_path)
+        instruments_service = InstrumentsService()
+        all_instruments = instruments_service.fetch_instruments()
+        instruments = get_requested_instruments(all_instruments, urns)
+        next_tasks = get_all_next_tasks(instruments, system_card)
+
+        for idx, lifecycle in enumerate(all_lifecycles):
+            click.echo("=" * 40)
+            click.echo(lifecycle)
+            click.echo("=" * 40)
+            for next_tasks_per_instrument in next_tasks:
+                for task in next_tasks_per_instrument["tasks_per_lifecycle"][idx]:
+                    click.echo(task.urn)
+    except Exception as error:
```

Comment:
Could we add more exceptions (So we can give better error messaging? Or want to do this later? 
example:
```
try:

except yaml.YAMLError:
    ... yaml could not be parsed exception 
Except as error:
   ...unknown exception
```

Reply:
Added your suggestion.